clients/pkg/promtail/client/manager.go

```diff
@@ -34,15 +39,24 @@ func NewManager(metrics *Metrics, logger log.Logger, maxStreams, maxLineSize int
 	if err != nil {
 		return nil, err
 	}
+
 	walWriter := wal.NewWriter(pWAL, logger)
 
+	watcherMetrics := wal.NewWatcherMetrics(reg)
+
 	var fake struct{}
 
 	if len(clientCfgs) == 0 {
 		return nil, fmt.Errorf("at least one client config should be provided")
 	}
 	clientsCheck := make(map[string]struct{})
+
 	clients := make([]Client, 0, len(clientCfgs))
+	watchers := make([]Stoppable, 0, len(clientCfgs))
+	// Instead of each write to that connects a watcher to a client keeping it's own mapping of decoded series, using
+	// a shared one
+	seriesCache := newSeriesCache()
```

Comment:
we have to be careful about this, the reason prometheus queue managers each have a watcher and cache is so that one remote write falling behind doesn't mean we can't clean up series records from the cache

the obvious downside to that is that we duplicate the caching for each queue manager (client in this case)

if we can come up with a nice _eventual_ clean up mechanism then we should be fine

Reply:
Yep, agree. Switching for to using a series cache per writeTo. We can used a shared later once the cleanup mechanism is more clear.